src/back/common/base.controller.spec.js

```diff
@@ -0,0 +1,289 @@
+'use strict';
+
+const BaseController = require('./base.controller'),
+    httpMocks = require('node-mocks-http'),
+    Q = require('q');
+
+let sut,
+    testModel;
+
+describe('ProfileController', () => {
+
+    beforeEach(() => {
+        testModel = function ModelStub() {
+        };
+        testModel.paginate = env.stub();
+        testModel.findById = env.stub();
+        testModel.prototype.save = env.stub();
+        testModel.findByIdAndUpdate = env.stub();
+        testModel.remove = env.stub();
+        sut = new BaseController(testModel);
+    });
+
+    describe('Method get', () => {
+
+        it('return expected object on get, if success result', (done) => {
+            const findPromise = Q.fcall(() => ([
+                { test: 'test' }
+            ]));
+            testModel.paginate.returns(findPromise);
+            const request = httpMocks.createRequest({
+                method: 'GET',
+                url: '/',
+                params: { id: 42 }
+            });
+            const response = httpMocks.createResponse();
+            sut.get(request, response)
+                .then(() => {
+                    response.statusCode.should.equal(200);
+                    const data = JSON.parse(response._getData());
+                    data.should.deep.equal([{
+                        test: 'test'
+                    }]);
+                    testModel.paginate.should.calledWith({}, {
```

Comment:
I would make these two assertions separate tests: one for `200` and the other for `page`, `limit` and other. It is easier to define test failing reason in one-assert-per-test structure.